### DISCUSSION TITLE
Webstorm IntelliSense

### DISCUSSION THREAD

**Discussion by cliqer** [windicss/windicss#136] [Ideas]
Is this a bug or am I missing something?

**Comment by cliqer** [comment]
@antfu on Webstorm autocomplete of classes does not work at all like it used to with old tailwind implementation.
![image](https://user-images.githubusercontent.com/3495993/110235512-49e38900-7f28-11eb-826f-4995165c4d83.png)

I'm also using pug and I assume webstorm needs to find the tailwind classes somewhere compiled to provide autocomplete.

Is there anything like IntelliSense for visual studio code that can work with webstorm?
Do I need to do something else or am I missing an option within the plugin?

Tried with the following options that didn't make any change:
```js
      transformCSS: 'pre',
      preflight: {
        enableAll: true
      }
```

**Comment by jdf221** [comment]
You can try leaving Tailwind as a dependency in your package.json. I still get autocomplete as long as it is a dependency.

Webstorm looks in nearest package.json for the dependency in order to know to use it. (https://youtrack.jetbrains.com/issue/WEB-42792#focus=Comments-27-4482485.0-0)

**Reply by ShermanTsang** [reply]
It's not work now.

**Reply by HeathHe** [reply]
@ShermanTsang Is there an up-to-date solution

**Reply by ShermanTsang** [reply]
No solution for now.

**Comment by antfu** [comment]
This is indeed on our plan, but we haven't got enough bandwidth to work on it yet. If you would love to see the WebStrom plugin for Windi CSS get started, please vote this comment, Thanks!

(also if you are interested in making one, we will be happy to make it official and you to join the team to work with us 🙌

**Reply by zhl154381001** [reply]
期待该插件

**Reply by difoi** [reply]
插件什么时候开发好...

**Reply by zhl154381001** [reply]
立项一年半了。。。。。。。。。。。

**Reply by itisl2220** [reply]
插件什么时候开发好...

**Reply by adoin** [reply]
Webstorm 太傲了

**Reply by vienpt** [reply]
need

**Reply by davodaslanifakor** [reply]
Thanks!

**Reply by Vi-jay** [reply]
nnnnnnnnnneeeeeeeeeeeeeddddddddddddddDDD!!!!!!!!

**Reply by dcl-lily** [reply]
等待+1

**Reply by anonvic** [reply]
11111111111

**Reply by jhoon2002** [reply]
need!!!!!

**Reply by suka233333** [reply]
gogogogo

**Reply by Viiprogrammer** [reply]
`+`

**Reply by shahensaifullah** [reply]
we need that.

**Reply by khashayarghajar** [reply]
We love to see the WebStrom plugin for Windi CSS

**Reply by itsdapi** [reply]
我太期待啦！！！速速做！

**Reply by JenuelDev** [reply]
This is really needed,,

**Reply by dylan0805** [reply]
This is really needed,,

**Reply by codemariosec** [reply]
I need it very much,thanks.

**Reply by Mrlilili** [reply]
Windi CSS is Sunsetting

https://windicss.org/posts/sunsetting.html

**Reply by KenLilyblade** [reply]
爱了

**Reply by tianjinjiaqiankeji** [reply]
爱了

**Reply by tianjinjiaqiankeji** [reply]
希望尽快推出webstorm插件

**Reply by lovinjs** [reply]
we need it

**Reply by JenuelDev** [reply]
a lot of people are now transitioning to webstorm, and to phpstorm, I think we need this one.

**Reply by zhong067310** [reply]
please

**Reply by suheIvy** [reply]
So when will you support it (*❦ω❦)

**Reply by wangjianqiangg** [reply]
2026年9月3日了，还会出吗

**Comment by AndrewBogdanovTSS** [comment]
You can use this Webstorm Tailwind extension until Windi support will be added https://plugins.jetbrains.com/plugin/15321-tailwind-css

**Reply by ghost** [reply]
this is a bad idea

**Reply by AndrewBogdanovTSS** [reply]
do you have a better one?

**Comment by kingwrcy** [comment]
Any plans on WebStorm plugin?

Really need it.

**Reply by 0x0a0d** [reply]
use tailwind.config.js instead of windi.config.js worked for me

**Reply by AndrewBogdanovTSS** [reply]
It doesn't provide full support though

**Reply by 0x0a0d** [reply]
yep, temp solution

**Reply by LLiuHuan** [reply]
2021-11-24

WebStorm Install the tailwind CSS plugins(https://plugins.jetbrains.com/plugin/15321-tailwind-css)

Then run

npm install --save-dev tailwindcss@npm:@tailwindcss/postcss7-compat@2.1.4
npm install --save-dev install --save-dev @tailwindcss/postcss7-compat2.1.4
or
yarn add -D tailwindcss@npm:@tailwindcss/postcss7-compat@2.1.4
yarn add -D @tailwindcss/postcss7-compat2.1.4

**Comment by bhvngt** [comment]
using `tailwind.config.js` hack does not seems to be working any more. Would be great to have a windicss specific plugin. I am switching over to `windicss` and so far it has worked out really smooth except for this `IDE support`. My current IDE is `intellij`. For various reasons `VS Code` is not an option for me.

**Comment by antick** [comment]
This is the only reason that I am not able to switch fully to windicss. I hope they get to work on it soon.

**Reply by creazy231** [reply]
same for me

**Reply by Blinks44** [reply]
same

**Reply by Alfred-Skyblue** [reply]
same for me

**Reply by chz** [reply]
same

**Reply by Xwudao** [reply]
same

**Reply by VladimirCores** [reply]
same

**Comment by mammadataei** [comment]
Hi, any news on this?

**Reply by fxjs** [reply]
Hi, any news on Webstorm IntelliSense?

**Comment by cdwmhcc** [comment]
Any news on this?

**Reply by GoogooFly** [reply]
hi pretty gril😀

**Comment by fishuke** [comment]
any update?

**Comment by jueinin** [comment]
any update？

**Comment by chz** [comment]
Any update?

**Comment by Shimada666** [comment]
Any update?

**Comment by tyaqing** [comment]
Any update?

**Comment by creazy231** [comment]
Really would like to see a working solution. We'd like to use WindiCSS all over our projects but without IntelliSense Support there is no way to do so 😞

**Comment by zhongjiabao** [comment]
any update?

**Comment by Chanzhaoyu** [comment]
Any news on this?

**Comment by MaxVilson** [comment]
Any update?

**Comment by chz** [comment]
**Any update ? ;)**

**Comment by D3SOX** [comment]
Guys, please stop spamming this with useless comments. There is an upvote button on the original post. I've subscribed to this to get actual updates and not just another `updates?` comment. Also if nobody works on this there won't be any updates :wink:

**Reply by wvffle** [reply]
I agree that this is annoying, though upvoting does not notify devs, while comments do.

**Comment by fishuke** [comment]
I know thats annoying thats what I am trying to do.

**Reply by D3SOX** [reply]
@fishuke @wvffle Do you really think somebody is going to implement it because there is a +1 comment every day? If they get annoyed, they can also just unsubscribe.
If you really need it, try implementing it yourself. I found https://github.com/jsbeckr/tailwind-idea as a starting point. Not sure about the license of the official JetBrains Tailwind plugin.

**Reply by fishuke** [reply]
I am reconsidering it now and i dont really need it lol. I started to remember classes without autocomplete. I am unsubscribing now. Thank you...

**Comment by iFTY-R** [comment]
Any update?

**Comment by blowsie** [comment]
Please upvote this ticket too, maybe it could be picked up on the jet brains side.
https://youtrack.jetbrains.com/issue/WEB-50504

**Comment by rhukster** [comment]
upvoted both places!

**Comment by fend25** [comment]
@antfu Hi!
First of all, thank you for WindiCSS!

Regarding this topic, devs from the IntelliJ ask: 
> Tailwind CSS plugin uses [Tailwind CSS Language Server](https://www.npmjs.com/package/@tailwindcss/language-server) under the hood. Do you think it may be used for Windi CSS as well?

Discussion:
https://youtrack.jetbrains.com/issue/WEB-50504

So, Anthony, is there any language server behind the WindiCSS? And if it is, is it's API compatible with the Tailwind CSS Language Server's one?

**Comment by nazar1ua** [comment]
Can we fork [Tailwind CSS Language Server](https://www.npmjs.com/package/@tailwindcss/language-server)? Or ask @tailwindlabs to source code and modify it for windicss. Also we can ask for jetbrains to give us souce code for tailwindcss plugin. Maybe I will can help with plugin for intellij idea.

**Reply by nazar1ua** [reply]
I think intellij plugin is very necessary, because it is very popular IDE

**Comment by DoveAz** [comment]
plz plz

**Comment by zhongjiabao** [comment]
please

**Comment by SirawichDev** [comment]
please do it

**Comment by y1324** [comment]
plz plz

**Comment by nazar1ua** [comment]
we need to fork this: https://github.com/tailwindlabs/tailwindcss-intellisense/tree/HEAD/packages/tailwindcss-language-server

**Reply by nazar1ua** [reply]
and then Jetbrains teem will create plugin for windi

**Comment by badlym** [comment]
现在我已经找到解决的办法了,就是添加windicss-analysis 这个插件这样就可以正确的提示了
https://github.com/windicss/windicss-analysis

**Reply by guohui666** [reply]
good idea

**Reply by sunpm** [reply]
> 现在我已经找到解决的办法了,就是添加windicss-analysis 这个插件这样就可以正确的提示了 https://github.com/windicss/windicss-analysis

然后 idea 就可以代码提示了吗

**Reply by adoin** [reply]
我试了不行

**Comment by anonvic** [comment]
When?

**Reply by AndrewBogdanovTSS** [reply]
Never I suppose

**Comment by reatlat** [comment]
🆙

**Comment by pgrzyb-rankomat** [comment]
And PHPStorm too.......

**Comment by sunpm** [comment]
两年过去了，会不会以后都不支持WebStorm

**Reply by adoin** [reply]
没人做咋办呢。。。

**Reply by sunpm** [reply]
可惜，vscode插件真丰富，我好羡慕

**Comment by zhl154381001** [comment]
已经放弃这个转tailwind了


| |
heling.zhang
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 2/21/2023 17:37 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [windicss/windicss] Webstorm IntelliSense (Discussion #136) |

没人做咋办呢。。。

**Reply by sunpm** [reply]
我也是因为这个原因，没有使用windicss和unocss

**Reply by adoin** [reply]
> 我也是因为这个原因，没有使用windicss和unocss

我还好，基本的样式都能猜出来，不熟练的再查咯、uno和windi都可以

**Comment by pzc666** [comment]
we need that.

**Comment by mkalus** [comment]
+1 here, too!

**Comment by zws-code** [comment]
webstorm 2023.1版本官方  已经支持tailwind插件了. 但是不支持 windicss和unocss,
但是可以通过安装tailwind 插件. 但是在项目中不引入, 添加提示功能.  因为windicss和unocss 完全兼容
npm install -D tailwindcss
npx tailwindcss init
<img width="634" alt="image" src="https://user-images.githubusercontent.com/19341509/232672552-c9dd0e8c-ad41-4c5f-9e5a-e1773b5d1d29.png">
_u_ 是我加的前缀. 可见 提示和预览都没问题

**Reply by adoin** [reply]
> webstorm 2023.1版本官方 已经支持tailwind插件了. 但是不支持 windicss和unocss, 但是可以通过安装tailwind 插件. 但是在项目中不引入, 添加提示功能. 因为windicss和unocss 完全兼容 npm install -D tailwindcss npx tailwindcss init <img alt="image" width="634" src="https://user-images.githubusercontent.com/19341509/232672552-c9dd0e8c-ad41-4c5f-9e5a-e1773b5d1d29.png"> _u_ 是我加的前缀. 可见 提示和预览都没问题

能凑合用和能用是两码事，能用和好用更是两码事啊

**Comment by MrZengCHN** [comment]
现在是2024年5月7日，继续等Webstorm的插件     ε=(´ο｀*)))

**Comment by Mrlilili** [comment]
Windicss can easily switch to Unocss, Unocss is supported by plugins.

**Comment by gzwxh** [comment]
加油，我们等您完成
